File: system-tests/tia-client-tia-mode-http/src/test/java/com/teamscale/tia/client/TiaClientTiaModeHttpSystemTest.java
package com.teamscale.tia.client;

import com.teamscale.report.testwise.model.TestInfo;
import com.teamscale.test.commons.SystemTestUtils;
import org.junit.jupiter.api.Test;
import testframework.CustomTestFramework;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * Runs the {@link CustomTestFramework} from src/main with our agent attached in tia-mode=http (the agent is configured
 * in this project's build.gradle). The custom test framework contains an integration with the tia-client. Asserts that
 * the resulting {@link TestInfo}s look as expected.
 * <p>
 * This ensures that our tia-client library doesn't break unexpectedly.
 */
public class TiaClientTiaModeHttpSystemTest {

	@Test
	public void systemTest() throws Exception {
		CustomTestFramework customTestFramework = new CustomTestFramework(SystemTestUtils.AGENT_PORT);
		customTestFramework.runTestsWithTia();

		assertThat(customTestFramework.testInfos.stream().map(SystemTestUtils::getCoverageString))
				.containsExactlyInAnyOrder("SystemUnderTest.java:4,13", "SystemUnderTest.java:4,8");
	}

}

File: system-tests/tia-client-tia-mode-http/src/test/kotlin/com/teamscale/tia/client/TiaClientTiaModeHttpSystemTest.kt
package com.teamscale.tia.client

import com.teamscale.test.commons.SystemTestUtils
import com.teamscale.test.commons.SystemTestUtils.coverage
import org.assertj.core.api.Assertions
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import testframework.CustomTestFramework

/**
 * Runs the [CustomTestFramework] from src/main with our agent attached in tia-mode=http (the agent is configured
 * in this project's build.gradle). The custom test framework contains an integration with the tia-client. Asserts that
 * the resulting [TestInfo]s look as expected.
 *
 *
 * This ensures that our tia-client library doesn't break unexpectedly.
 */
class TiaClientTiaModeHttpSystemTest {
	@Test
	@Throws(Exception::class)
	fun systemTest() {
		val customTestFramework = CustomTestFramework(SystemTestUtils.AGENT_PORT)
		customTestFramework.runTestsWithTia()

		assertThat(customTestFramework.testInfos.map { it.coverage })
			.containsExactlyInAnyOrder("SystemUnderTest.kt:4,6", "SystemUnderTest.kt:4,9")
	}
}
